addons/point_of_sale/models/pos_session.py

```diff
@@ -701,14 +703,22 @@ def _get_stock_output_vals(self, out_account, amount, amount_converted):
         return self._credit_amounts(partial_args, amount, amount_converted)
 
     def _get_statement_line_vals(self, statement, receivable_account, amount):
-        return {
-            'date': fields.Date.context_today(self),
+        vals = {
             'amount': amount,
-            'name': self.name,
+            'payment_ref': self.name,
             'statement_id': statement.id,
-            'account_id': receivable_account.id,
+            'journal_id': statement.journal_id.id,
         }
 
+        # Override the temporary account to set directly the receivable one in order to perform the reconciliation
+        # faster without using the statement reconciliation.
+        line_ids_vals = self.env['account.bank.statement.line']._prepare_move_line_default_vals(vals)
+        for line_vals in line_ids_vals:
+            if line_vals['account_id'] == statement.journal_id.suspense_account_id.id:
+                line_vals['account_id'] = receivable_account.id
+        vals['line_ids'] = [(0, 0, line_vals) for line_vals in line_ids_vals]
+        return vals
+
```

Comment:
Before this task, the PoS was creating payments through the statement reconciliation only for cash journals. Now, to avoid an overhead with the statement reconciliation, I set directly the right receivable account instead the suspense account to perform the reconciliation of move lines directly.